python-main.js

```diff
@@ -1213,22 +1241,7 @@ function web_editor(config) {
             // console.log("Time taken to flash: " + Math.floor(timeTaken) + " minutes, "
             //           + Math.ceil((timeTaken - Math.floor(timeTaken)) * 60) + " seconds");
         })
-        .catch(function(err) {
-            console.log("Error flashing: " + err);
-            $("#flashing-overlay-container").css("display", "flex");
-            $("#webusb-flashing-progress").css("display", "none");
-
-            // If micro:bit does not support dapjs
-            if (err.message === "No valid interfaces found.") {
-                $("#flashing-overlay-error").html('<div>' + err + '</div><div>'+ config["translate"]["webusb"]["err"]["update-req"] +'</div><a href="#" onclick="flashErrorClose()">'+ config["translate"]["webusb"]["close"] +'</a>');
-                return;
-            } else if(err.message === "Unable to claim interface.") {
-                $("#flashing-overlay-error").html('<div>'+ config["translate"]["webusb"]["err"]["clear-connect"] +'</div><a href="#" onclick="flashErrorClose()">'+ config["translate"]["webusb"]["close"] +'</a>');
-                return;
-            }
-
-            $("#flashing-overlay-error").html('<div>' + err + '</div><div>'+ config["translate"]["webusb"]["err"]["restart-microbit"] +'</div><a href="#" onclick="flashErrorClose()">'+ config["translate"]["webusb"]["close"] +'</a>');
-        });
+        .catch(webubsErrorHandler);
```

Comment:
By pressing the reset button during flash I have also bee able to trigger the following issue:

![image](https://user-images.githubusercontent.com/29712657/63354487-6e830a80-c35c-11e9-8f60-2f5bff2cc444.png)

The exception not caught means that this handler is not executed and the editor page get's locked in this modal:
![image](https://user-images.githubusercontent.com/29712657/63354539-88245200-c35c-11e9-865b-632630b43334.png)

Reply:
👍

Reply:
To catch this I'm going to have to add a generic handler, that will catch all unhandled rejections in dapjs

When I catch it do I want to call `doDisconnect`? I think this can also occur while using the REPL, and isn't always fatal.. which would cause the REPL to disconnect unnecessarily

```
window.addEventListener("unhandledrejection", function(promiseRejectionEvent) { 
    // handle error here, for example log   
});
```

Reply:
Yeah, we need to investigate why that happens. Specially since MakeCode probably doesn't see the same issues.
For the interim, maybe add and remove the handler at the beginning and end of the flashing process?

Reply:
Done